Source/WebKit/UIProcess/Cocoa/XPCConnectionTerminationWatchdog.mm

```diff
@@ -41,13 +45,19 @@
     : m_xpcConnection(process.connection()->xpcConnection())
     , m_watchdogTimer(RunLoop::main(), this, &XPCConnectionTerminationWatchdog::watchdogTimerFired)
     , m_assertion(ProcessAndUIAssertion::create(process, "XPCConnectionTerminationWatchdog"_s, ProcessAssertionType::Background))
+    , m_process(process)
 {
     m_watchdogTimer.startOneShot(interval);
 }
     
 void XPCConnectionTerminationWatchdog::watchdogTimerFired()
 {
+#if USE(EXTENSIONKIT)
+    if (m_process->extensionProcess())
```

Comment:
We should store the value returned by `extensionProcess()` in a local to avoid calling the function twice.

Reply:
Should we RELEASE_LOG_ERROR in the else case? Since it would mean we fail to terminate the hung process.